sqrl-tools/sqrl-packager/src/main/java/com/datasqrl/packager/Packager.java

```diff
@@ -251,29 +218,91 @@ private boolean retrieveDependency(Path buildDir, NamePath packagePath, Dependen
     return repository.retrieveDependency(targetPath, dependency);
   }
 
-  public void cleanUp() {
-    try {
-      Path buildDir = rootDir.resolve(BUILD_DIR_NAME);
-      if (Files.exists(buildDir)) {
-        FileUtils.deleteDirectory(buildDir.toFile());
-      }
-    } catch (IOException e) {
-      throw new IllegalStateException("Could not read or write files on local file-system", e);
-    }
-  }
-
   public static Path copyRelativeFile(Path srcFile, Path srcDir, Path destDir) throws IOException {
     return copyFile(srcFile, destDir, srcDir.relativize(srcFile));
   }
 
   public static Path copyFile(Path srcFile, Path destDir, Path relativeDestPath)
       throws IOException {
-    Preconditions.checkArgument(Files.isRegularFile(srcFile), "Is not a file: {}", srcFile);
+    Preconditions.checkArgument(Files.isRegularFile(srcFile), "Is not a file: %s", srcFile);
     Path targetPath = canonicalizePath(destDir.resolve(relativeDestPath));
     if (!Files.exists(targetPath.getParent())) {
       Files.createDirectories(targetPath.getParent());
     }
     Files.copy(srcFile, targetPath, StandardCopyOption.REPLACE_EXISTING);
     return targetPath;
   }
+
+  public void postprocess(CompilerResult result, Path targetDir, Optional<Path> mountDirectory,
+      String[] profiles) {
+    List.of(new DockerPostprocessor(), new FlinkPostprocessor())
+        .forEach(p->p.process(new ProcessorContext(buildDir, targetDir, result, mountDirectory, profiles)));
+  }
+
+  @SneakyThrows
+  public static Path writeEngineConfig(Path rootDir, SqrlConfig config) {
+    Path enginesFile = Files.createTempFile(rootDir, "package-engines", ".json");
+    File file = enginesFile.toFile();
+    file.deleteOnExit();
+
+    config.toFile(enginesFile, true);
+    return enginesFile;
+  }
+
+  public static Optional<List<Path>> findPackageFile(Path rootDir, List<Path> packageFiles) {
+    if (packageFiles.isEmpty()) {
+      Path defaultPkg = rootDir.resolve(DEFAULT_PACKAGE);
+      if (Files.isRegularFile(defaultPkg)) {
+        return Optional.of(List.of(defaultPkg));
+      } else {
+        return Optional.empty();
+      }
+    } else {
+      return Optional.of(packageFiles);
+    }
+  }
+
+  public static SqrlConfig createDockerConfig(ErrorCollector errors) {
```

Comment:
Why do we still need to hard-code this in here? Is it possible to make this a profile that we ship with DataSQRL?

Reply:
this is 'default' config for the compile and validate targets when no other package.json is specified. It's awkwardly named, but we can further modularize it out in the future.